bin/database.py

```diff
@@ -51,6 +53,44 @@ def confirm_schema_match():
     else:
         sys.exit(0)
 
+def getMonotonicTime():
+    # http://stackoverflow.com/a/7424304
+    return os.times()[4]
+
+def process_cursor(cursor, closure):
+    """
+    Given an iterable (say, a mongo cursor) and a closure, call that closure in parallel over the iterable.
+    Call order is undefinied. Currently launches N python process workers, where N is the number of vcpu cores.
+
+    Useful for upgrades that need to touch each document in a database, and don't need an iteration order.
```

Comment:
Maybe just add a comment that the expected return value of the closure is either `True` or a `str` error?